api/v1/Batch.php

```diff
@@ -33,66 +31,49 @@ public function __construct(string $method = '')
     }
 
     /**
-     * @apiGroup Batch
      * @throws Exception
-     * @api {POST} /batch/startbatch StartBatch
-     * @apiDescription Starts a new batch
-     *
-     * @apiParam {int} $country_id id of country/team for which the batch was started
-     * @apiParam {int} $user_id id of user for which the batch was started
-     *
-     * @apiSuccess {int} batch_id Batch ID used to identify this batch.
-     * @noinspection PhpUnused
      */
-    // phpcs:ignore PSR1.Methods.CamelCapsMethodName.NotCamelCaps
-    public function StartBatch(int $country_id, int $user_id)
+    private function startBatch(int $countryId, int $userId, string $batchGuid): int
     {
-        return $this->getDatabase()->query(
-            "INSERT INTO api_batch(api_batch_country_id, api_batch_user_id) VALUES (?, ?)",
-            array($country_id, $user_id),
+        $id = $this->getDatabase()->query(
```

Comment:
was this what you were talking about this morning: the face that this does not use the async DB connection, but the separate old-skool DB connection? I wouldn't worry about it too much, at least not for RC1.

Reply:
Ok to explain this, let's go a bit into history. 

Prehistory: :)
So 2 weeks ago we did a separate StartBatch , AddToBatch and ExecuteBatch right? And we did not see any 10 sec timeouts that I remember. So basically this means that each could be processed within a 10 sec timeframe of an api call.  Basically they all did a single insert (also the ExecuteBatch when call with async enabled).

Now
Then I combined them into a single ExecuteBatch , of course causing a more heavy single api call because of it, because now it has to do a lot of insert in a single call. If there are 60 AddToBatch calls ... then there are also that amounts of inserts. So the 10 sec threshold might be reached. 

So I added transactions , and tested that . So I think the 10 sec  should not be possible anymore.